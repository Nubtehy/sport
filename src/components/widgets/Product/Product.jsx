import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'components/controls';
import {
  ProductContainer, ProductName, ProductPriceBlock, ProductPrice,
} from './Product.styled';

const displayName = 'Product';

const propTypes = {
  name: PropTypes.string.isRequired,
  handleAddToCart: PropTypes.func.isRequired,
};

function Product({
  name, image, handleAddToCart, price,
}) {
  return (
    name && (
      <ProductContainer data-name={name}>
        <img src={image} alt="" />
        <ProductName>
          {name}
        </ProductName>
        <ProductPriceBlock>
          <ProductPrice>
            {price}
            {' '}
          $
          </ProductPrice>
          <Button accent onClick={handleAddToCart}>
            Add to cart
          </Button>
        </ProductPriceBlock>

      </ProductContainer>
    )
  );
}

Product.displayName = displayName;
Product.propTypes = propTypes;

export default Product;
