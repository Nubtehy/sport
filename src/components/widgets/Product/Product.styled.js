import styled from 'styled-components';

export const ProductWrapper = styled.div`
  display: flex;
  justify-content: space-around;
`;

export const ProductContainer = styled.div`
  flex: 0 0 auto;
  width: 250px;
  position: relative;
  padding: 18px;
  border: 1px solid #e1e7ec;
  border-radius: 7px;
  background-color: #fff;
  margin: 20px 0;
  img {
    width: 100%;
  }
`;
export const ProductName = styled.div`
  font-family: 'Poppins', sans-serif;
`;

export const ProductPrice = styled.div`

`;
export const ProductPriceBlock = styled.div`
  display: flex;
  justify-content: space-between;
`;
