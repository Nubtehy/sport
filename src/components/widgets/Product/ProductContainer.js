import {
  compose, setDisplayName, withHandlers, setPropTypes,
} from 'recompose';
import PropTypes from 'prop-types';
import Product from './Product';

export const enhance = compose(
  setDisplayName('ProductContainer'),
  setPropTypes({
    id: PropTypes.number.isRequired,
    AddToCart: PropTypes.func.isRequired,
  }),
  withHandlers({
    handleAddToCart: ({ id, AddToCart }) => () => AddToCart(id),
  }),
);

export default enhance(Product);
