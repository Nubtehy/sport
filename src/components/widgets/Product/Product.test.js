import React from 'react';
import { shallow } from 'enzyme';
import Product from './Product';

const testProps = {
  name: 'Product 1',
  image: '',
  handleAddToCart: jest.fn(),
  price: 212,
};
describe('Given the CartItem component', () => {
  describe('when the component is rendered', () => {
    const component = shallow(<Product {...testProps} />);
    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
