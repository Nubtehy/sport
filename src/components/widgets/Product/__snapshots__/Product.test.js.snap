// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CartItem component when the component is rendered should render the component with provided props 1`] = `
<Productstyled__ProductContainer
  data-name="Product 1"
>
  <img
    alt=""
    src=""
  />
  <Productstyled__ProductName>
    Product 1
  </Productstyled__ProductName>
  <Productstyled__ProductPriceBlock>
    <Productstyled__ProductPrice>
      212
       
      $
    </Productstyled__ProductPrice>
    <WithStyle(Button)
      accent={true}
      name="add-to-cart"
      onClick={[MockFunction]}
    >
      Add to cart
    </WithStyle(Button)>
  </Productstyled__ProductPriceBlock>
</Productstyled__ProductContainer>
`;
